### ISSUE TITLE
I can't connect to macronix 25l6406e.

### ISSUE THREAD

**Issue by csolg** [lukier/linux-hi3518#1]
Hello,

Please, help me. I can't connect to Macronix 25L6406E.

here log:

```
Uncompressing Linux... done, booting the kernel.
[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Linux version 4.4.0-rc4 (csolg@hi3518) (gcc version 6.1.0 (Buildroot 2016.05-g8d2785c-dirty) ) #2 Tue Aug 2 00:58:55 VLAT 2016
[    0.000000] CPU: ARM926EJ-S [41069265] revision 5 (ARMv5TEJ), cr=0005317f
[    0.000000] CPU: VIVT data cache, VIVT instruction cache
[    0.000000] Machine model: Hisilicon Hi3518 Camera
[    0.000000] Memory policy: Data cache writeback
[    0.000000] Built 1 zonelists in Zone order, mobility grouping on.  Total pages: 10160
[    0.000000] Kernel command line: mem=40M console=ttyAMA0,115200n8 mtdparts=hi_sfc:256k(uboot)ro,256k(uboot-env),3m(kernel),-(rootfs) root=/dev/mtdblock3 hieth.mdioifu=0 hieth.mdioifd=0 hieth.phy2
[    0.000000] PID hash table entries: 256 (order: -2, 1024 bytes)
[    0.000000] Dentry cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Inode-cache hash table entries: 4096 (order: 2, 16384 bytes)
[    0.000000] Memory: 33200K/40960K available (5140K kernel code, 204K rwdata, 1488K rodata, 160K init, 261K bss, 7760K reserved, 0K cma-reserved, 0K highmem)
[    0.000000] Virtual kernel memory layout:
[    0.000000]     vector  : 0xffff0000 - 0xffff1000   (   4 kB)
[    0.000000]     fixmap  : 0xffc00000 - 0xfff00000   (3072 kB)
[    0.000000]     vmalloc : 0xc3000000 - 0xff800000   ( 968 MB)
[    0.000000]     lowmem  : 0xc0000000 - 0xc2800000   (  40 MB)
[    0.000000]     pkmap   : 0xbfe00000 - 0xc0000000   (   2 MB)
[    0.000000]     modules : 0xbf000000 - 0xbfe00000   (  14 MB)
[    0.000000]       .text : 0xc0008000 - 0xc06816e4   (6630 kB)
[    0.000000]       .init : 0xc0682000 - 0xc06aa000   ( 160 kB)
[    0.000000]       .data : 0xc06aa000 - 0xc06dd068   ( 205 kB)
[    0.000000]        .bss : 0xc06dd068 - 0xc071e628   ( 262 kB)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS:16 nr_irqs:32 32
[    0.000000] VIC @c3000000: id 0x00641190, vendor 0x41
[    0.000000] LUKIER-ClkHi3518 Initializing hi3518-clock node <NULL> clock /clocks/clock@20030000
[    0.000000] clocksource: arm,sp804: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604467 ns
[    0.000026] sched_clock: 32 bits at 100MHz, resolution 10ns, wraps every 21474836475ns
[    0.000966] Console: colour dummy device 80x30
[    0.001036] Calibrating delay loop... 218.72 BogoMIPS (lpj=1093632)
[    0.069962] pid_max: default: 32768 minimum: 301
[    0.070239] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.070293] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.072272] CPU: Testing write buffer coherency: ok
[    0.073379] Setting up static identity map for 0x80008400 - 0x80008458
[    0.081132] devtmpfs: initialized
[    0.094585] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.095066] pinctrl core: initialized pinctrl subsystem
[    0.096590] NET: Registered protocol family 16
[    0.099262] DMA: preallocated 256 KiB pool for atomic coherent allocations
[    0.129052] Serial: AMBA PL011 UART driver
[    0.129828] 20080000.uart: ttyAMA0 at MMIO 0x20080000 (irq = 37, base_baud = 0) is a PL011 rev2
[    0.414747] console [ttyAMA0] enabled
[    0.419645] uart-pl011 20090000.uart: could not find pctldev for node /soc/pinmux@200F0000/pinmux_uart1_pins, deferring probe
[    0.431182] uart-pl011 200a0000.uart: could not find pctldev for node /soc/pinmux@200F0000/pinmux_uart2_pins, deferring probe
[    0.468788] pl08xdmac 100d0000.dma-controller: initialized 8 virtual memcpy channels
[    0.476766] pl08xdmac 100d0000.dma-controller: initialized 12 virtual slave channels
[    0.492801] pl08xdmac 100d0000.dma-controller: DMA: PL080 rev0 at 0x100d0000 irq 46
[    0.501938] SCSI subsystem initialized
[    0.505985] ssp-pl022 200c0000.spi0: could not find pctldev for node /soc/pinmux@200F0000/pinmux_spi0_pins, deferring probe
[    0.517349] ssp-pl022 200e0000.spi1: could not find pctldev for node /soc/pinmux@200F0000/pinmux_spi1_pins, deferring probe
[    0.530184] usbcore: registered new interface driver usbfs
[    0.535999] usbcore: registered new interface driver hub
[    0.541648] usbcore: registered new device driver usb
[    0.547377] Linux video capture interface: v2.00
[    0.555881] clocksource: Switched to clocksource arm,sp804
[    0.607942] NET: Registered protocol family 2
[    0.614828] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    0.622059] TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
[    0.628612] TCP: Hash tables configured (established 1024 bind 1024)
[    0.635296] UDP hash table entries: 256 (order: 0, 4096 bytes)
[    0.641364] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
[    0.648501] NET: Registered protocol family 1
[    0.654156] RPC: Registered named UNIX socket transport module.
[    0.660233] RPC: Registered udp transport module.
[    0.664959] RPC: Registered tcp transport module.
[    0.669752] RPC: Registered tcp NFSv4.1 backchannel transport module.
[    0.679847] futex hash table entries: 256 (order: -1, 3072 bytes)
[    0.729339] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.740217] NFS: Registering the id_resolver key type
[    0.745396] Key type id_resolver registered
[    0.749730] Key type id_legacy registered
[    0.753857] nfs4filelayout_init: NFSv4 File Layout Driver Registering...
[    0.762775] fuse init (API version 7.23)
[    0.784244] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 252)
[    0.791880] io scheduler noop registered
[    0.795956] io scheduler deadline registered
[    0.800403] io scheduler cfq registered (default)
[    0.806365] pinctrl-single 200f0000.pinmux: 95 pins at pa c3016000 size 380
[    0.814712] pl061_gpio 20140000.gpio: PL061 GPIO chip @0x20140000 registered
[    0.822891] pl061_gpio 20150000.gpio: PL061 GPIO chip @0x20150000 registered
[    0.831063] pl061_gpio 20160000.gpio: PL061 GPIO chip @0x20160000 registered
[    0.839225] pl061_gpio 20170000.gpio: PL061 GPIO chip @0x20170000 registered
[    0.847389] pl061_gpio 20180000.gpio: PL061 GPIO chip @0x20180000 registered
[    0.855544] pl061_gpio 20190000.gpio: PL061 GPIO chip @0x20190000 registered
[    0.863719] pl061_gpio 201a0000.gpio: PL061 GPIO chip @0x201a0000 registered
[    0.871861] pl061_gpio 201b0000.gpio: PL061 GPIO chip @0x201b0000 registered
[    0.880076] pl061_gpio 201c0000.gpio: PL061 GPIO chip @0x201c0000 registered
[    0.888222] pl061_gpio 201d0000.gpio: PL061 GPIO chip @0x201d0000 registered
[    0.896384] pl061_gpio 201e0000.gpio: PL061 GPIO chip @0x201e0000 registered
[    0.904507] pl061_gpio 201f0000.gpio: PL061 GPIO chip @0x201f0000 registered
[    1.203747] Serial: AMBA driver
[    1.262325] brd: module loaded
[    1.291638] loop: module loaded
[    1.302752] nbd: registered device at major 43
[    1.355736] libphy: Fixed MDIO Bus: probed
[    1.366999] hieth 10090000.ether: using random MAC address 2e:41:ca:27:d5:0d
[    1.376459] hieth 10090000.ether: eth0: at c34e0000, IRQ 44 MAC: 2e:41:ca:27:d5:0d
[    1.385078] libphy: hieth_mii_bus: probed
[    1.428509] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    1.435152] ehci-platform: EHCI generic platform driver
[    1.440988] ehci-platform 100b0000.ehci: EHCI Host Controller
[    1.447012] ehci-platform 100b0000.ehci: new USB bus registered, assigned bus number 1
[    1.458248] ehci-platform 100b0000.ehci: can't setup: -110
[    1.463799] ehci-platform 100b0000.ehci: USB bus 1 deregistered
[    1.469947] ehci-platform: probe of 100b0000.ehci failed with error -110
[    1.477115] ohci_hcd: USB 1.1 'Open' Host Controller (OHCI) Driver
[    1.483402] ohci-platform: OHCI generic platform driver
[    1.489205] ohci-platform 100a0000.ohci: Generic Platform OHCI controller
[    1.496270] ohci-platform 100a0000.ohci: new USB bus registered, assigned bus number 1
[    1.504575] ohci-platform 100a0000.ohci: irq 48, io mem 0x100a0000
[    1.565948] ohci-platform 100a0000.ohci: init err (00000000 0000)
[    1.572079] ohci-platform 100a0000.ohci: can't start
[    1.577342] ohci-platform 100a0000.ohci: startup error -75
[    1.582897] ohci-platform 100a0000.ohci: USB bus 1 deregistered
[    1.589092] ohci-platform: probe of 100a0000.ohci failed with error -75
[    1.597536] mousedev: PS/2 mouse device common for all mice
[    1.604702] i2c /dev entries driver
[    1.615851] sp805-wdt 20040000.watchdog: registration successful
[    1.622696] sdhci: Secure Digital Host Controller Interface driver
[    1.629008] sdhci: Copyright(c) Pierre Ossman
[    1.633430] Synopsys Designware Multimedia Card Interface Driver
[    1.640116] dw_mmc 10020000.sdio: num-slots property not found, assuming 1 slot is available
[    1.648727] dw_mmc 10020000.sdio: fifo-depth property not found, using value of FIFOTH register as default
[    1.658564] dw_mmc 10020000.sdio: IDMAC supports 32-bit address mode.
[    1.665168] dw_mmc 10020000.sdio: Using internal DMA controller.
[    1.671424] dw_mmc 10020000.sdio: Version ID is 210a
[    1.676612] dw_mmc 10020000.sdio: DW MMC controller at irq 50,32 bit host data width,256 deep fifo
[    1.685741] dw_mmc 10020000.sdio: No vmmc regulator found
[    1.691264] dw_mmc 10020000.sdio: No vqmmc regulator found
[    1.736063] dw_mmc 10020000.sdio: 1 slots initialized
[    1.741441] sdhci-pltfm: SDHCI platform and OF driver helper
[    1.748538] usbcore: registered new interface driver usbhid
[    1.754140] usbhid: USB HID core driver
[    1.758883] ipip: IPv4 over IPv4 tunneling driver
[    1.769560] NET: Registered protocol family 10
[    1.779441] sit: IPv6 over IPv4 tunneling driver
[    1.786779] NET: Registered protocol family 17
[    1.791400] NET: Registered protocol family 15
[    1.796315] 8021q: 802.1Q VLAN Support v1.8
[    1.800770] Key type dns_resolver registered
[    1.813601] 20090000.uart: ttyAMA1 at MMIO 0x20090000 (irq = 37, base_baud = 0) is a PL011 rev2
[    1.824184] 200a0000.uart: ttyAMA2 at MMIO 0x200a0000 (irq = 57, base_baud = 0) is a PL011 rev2
[    1.834460] ssp-pl022 200c0000.spi0: ARM PL022 driver, device ID: 0x00041022
[    1.841842] ssp-pl022 200c0000.spi0: mapped registers from 0x200c0000 to c3600000
[    1.851141] ssp-pl022 200e0000.spi1: ARM PL022 driver, device ID: 0x00041022
[    1.858483] ssp-pl022 200e0000.spi1: mapped registers from 0x200e0000 to c3620000
[    1.868518] hctosys: unable to open rtc device (rtc0)
[    1.875769] uart-pl011 20080000.uart: no DMA platform data
[    1.882745] Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
[    1.891032] CPU: 0 PID: 1 Comm: swapper Not tainted 4.4.0-rc4 #2
[    1.897044] Hardware name: Hisilicon Hi3516/Hi3518 (Flattened Device Tree)
[    1.904003] [<c000f1e0>] (unwind_backtrace) from [<c000d16c>] (show_stack+0x10/0x14)
[    1.911795] [<c000d16c>] (show_stack) from [<c0062588>] (panic+0x80/0x1e0)
[    1.918719] [<c0062588>] (panic) from [<c0683190>] (mount_block_root+0x178/0x22c)
[    1.926245] [<c0683190>] (mount_block_root) from [<c0683434>] (mount_root+0xf4/0x120)
[    1.934112] [<c0683434>] (mount_root) from [<c06835b0>] (prepare_namespace+0x150/0x1b0)
[    1.942152] [<c06835b0>] (prepare_namespace) from [<c0682db0>] (kernel_init_freeable+0x180/0x1cc)
[    1.951072] [<c0682db0>] (kernel_init_freeable) from [<c04e6b00>] (kernel_init+0x8/0xe8)
[    1.959202] [<c04e6b00>] (kernel_init) from [<c000a450>] (ret_from_fork+0x14/0x24)
[    1.966791] ---[ end Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
[   82.695956] random: nonblocking pool is initialized
```

**Comment by lukier** [comment]
This is an abandoned project. I've enabled the peripherals that are standard (from ARM PrimeCell or Synposys) and have kernel drivers already, then I worked on the Ethernet driver (porting the original HiSilicon one) to boot over NFS, but then I dropped the project as the NAND or SPI Flash peripherals seem custom and I didn't bother to write drivers for them. And then in the end the important stuff (video processing, compression etc) are completely binary and mostly undocumented in the datasheet anyway. It would be a massive effort to write drivers for this chip without HiSilicon's help.
